packages/ui/components/form/inputs/HintOrErrors.tsx

```diff
@@ -1,106 +1,95 @@
 import type { FieldValues } from "react-hook-form";
 import { useFormContext } from "react-hook-form";
 
-import { Check, Circle, Info, X } from "../../icon";
+import { classNames } from "@calcom/lib";
+
+import { Check, Circle, X } from "../../icon";
 
 type hintsOrErrorsProps = {
   hintErrors?: string[];
   fieldName: string;
   t: (key: string) => string;
+  hideIfEmpty?: boolean;
 };
 
+function CustomErrors({ hintErrors, fieldName, t }: hintsOrErrorsProps) {
+  const methods = useFormContext();
+  const fieldErrors = methods.formState.errors[fieldName];
+
+  if (!fieldErrors) return null;
+
+  return (
+    <div className="text-gray text-default mt-2 flex items-center text-sm">
+      <ul className="ml-2">
+        {Object.keys(fieldErrors).map((key: string) => {
+          return (
+            <li key={key} className="text-blue-700">
+              {t(`${fieldName}_hint_${key}`)}
+            </li>
+          );
+        })}
+      </ul>
+    </div>
+  );
+}
+
 export function HintsOrErrors<T extends FieldValues = FieldValues>({
   hintErrors,
   fieldName,
   t,
+  hideIfEmpty = false,
 }: hintsOrErrorsProps) {
   const methods = useFormContext() as ReturnType<typeof useFormContext> | null;
   /* If there's no methods it means we're using these components outside a React Hook Form context */
   if (!methods) return null;
-  const { formState } = methods;
+
+  if (!methods && hintErrors?.length) {
+    throw new Error("InputField is expecting hint errors but is not use within a RHF context");
+  }
+
+  const { formState, watch } = methods;
+  const hasBeenTouched = watch(fieldName);
```

Comment:
I think it's not identifying if the input has been changed/touched. It is just returning the value.